### PR TITLE
Grammar generator

### PR DESCRIPTION
**Pull request by ldez**
- [x] merge PR #58 and #60 
- [x] split the grammars file into multiple files (partials grammars)
- [x] create a master grammar
- [x] create a grammars generator (use master grammar and partials grammars)
- [x] create a grammars generator for the code blocks (Asciidoctor and Markdown) 
- [x] Document generation system. (and GitHub PR template)

I've created a package for gitignore's like file with a small generator (https://github.com/ldez/atom-language-ignore).
I'm going to use this as starting point.

### CODE REVIEW

**Comment by ldez** [comment]
@nicorikken you can try the new system. :smile:

**Comment by ldez** [comment]
@nicorikken and @mojavelinux It is ready!

I add the change from the #67 (fix #65)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+*Remove the previous 'Preamble' section upon you have read it.*
+
+## Description
+
+Your description.
+
+## Syntax example
+
+Your syntax exemple.
```
Comment (nicorikken) [inline .github/PULL_REQUEST_TEMPLATE.md]: Typo?

File: grammars/repositories/languages.cson [hunk 2]
```diff
@@ -0,0 +1,79 @@
+[
```
Comment (nicorikken) [inline grammars/repositories/languages.cson]: Maybe order this alphabetically?

**Comment by nicorikken** [comment]
Whow, this is amazing stuff!!! :+1: :beers: :clap: Have to reorient myself somewhat though. 
Some remarks:
- [ ] Order the languages alphabetically in `grammars/repositories/languages`
- [ ] Compile with the latest grammar, to include the `table` and `various-blocks`
- [ ] Document the `season` dependency somewhere (at least, that was the one that I was missing) Maybe just `npm install` ?

As this is quite a breaking change, I opt for getting this merge next, and implement/rewrite all other PR's on top of this PR.

**Comment by ldez** [comment]
I don't understand:
> Compile with the latest grammar, to include the table and various-blocks

For installing the dependencies do: `apm install`

File: grammars/repositories/languages.cson (hunk 2 above)
Comment (ldez) [inline grammars/repositories/languages.cson]: Done!

**Comment by nicorikken** [comment]
> For installing the dependencies do: `apm install`

I didn't know that actually... :disappointed: :+1:

**Comment by ldez** [comment]
- [x] Order the languages alphabetically in grammars/repositories/languages
- [x] Document the `season` dependency somewhere

**Comment by nicorikken** [comment]
Somehow when I recompiled the grammar last time, it did result in the two sections added on top of the checkout. Now it seems to be fine. I'm up for a merge!

**Comment by ldez** [comment]
You can also close  #36 and #67  :wink:

**Comment by ldez** [comment]
@nicorikken  can you make a release?

use:
- `apm publish major`: 0.12.0 -> 1.0.0
- or `apm publish minor`: 0.12.0 -> 0.13.0

Those commands do automatically a tag on Git.

**Comment by nicorikken** [comment]
Merged! Some improvements here and there will probably still be required, but now we have a new structure to base the reset of the development on.

**Comment by nicorikken** [comment]
@ldez I was aiming to get my bold fix incorporated, and maybe rewrite my previous italic PR (which is too greedy). What do you think?

**Comment by ldez** [comment]
I think it's releasing time !

I am looking forward to use all the new feature outside developer mode.
There are already a lot of changes since the last release.

But if you want to add the bold fix, I can wait :wink:

**Comment by nicorikken** [comment]
Alright, working on it! Same here, can't wait to share these improvements with my colleagues in a regular package update.

**Comment by mojavelinux** [comment]
I think we should move properties-lang.cson into a subdirectory, perhaps "contrib" or "alt". I think the visibility of the file is too high when adjacent to asciidoctor-grammar.cson.

**Comment by nicorikken** [comment]
Good catch. I'v opened it up accidentally as well.

**Comment by ldez** [comment]
we cannot do that, the file grammar must be in a first level of `grammars` folder.

**Comment by mojavelinux** [comment]
I see. Can it be named `language-properties.cson` instead of `properties-lang.cson`?

**Comment by ldez** [comment]
yes :+1:

**Comment by ldez** [comment]
Fix with  #101

**Comment by mojavelinux** [comment]
\o/

**Pull request merged**